ush/wafs_grib2_0p25_blending.sh

```diff
@@ -0,0 +1,145 @@
+    # Blending ecflow run is triggered by the finishing of the upstream JWAFS_GRIB2_0P25 f048
+    #  - In real time ecflow, blending starts at T+4:30, unblended upstream products are all available.
+    #    It doesn't need to sleep/wait
+    #  - In non-real time ecflow, blending has no time tigger, unblended f048 is not guaranteed to be the last finished one.
+    #    It needs to sleep/wait. All upstream jobs can be finished within 60 seconds when f048 is finished.
+    #
+    # Standalone blending doesn't need to sleep/wait since unblended upstream grib2_0p25 runs standalone first.
+    sleep 60
+    if [[ ! -f "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2.idx" ]]; then
+	echo "WARNING: missing US unblended data - ${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2"
+	echo "Missing ${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2" > ../missing_us_files.$fhr
+	MISSING_US_WAFS="YES"
+    fi
+fi
+
+##########################
+# Blending or unblended
+##########################
+if [[ "${MISSING_UK_WAFS}" == "YES" ]] && [[ "${MISSING_US_WAFS}" == "YES" ]]; then
+    cat ../missing_uk_files.$fhr ../missing_us_files.$fhr > ../no_blending_files.$fhr
+    rm ../missing_uk_files.$fhr ../missing_us_files.$fhr
+elif [[ "${MISSING_UK_WAFS}" == "YES" ]]; then
+    echo "turning back on dbn alert for unblended US WAFS product"
+    # Avoid duplicate dbn_alert of unblended grib2 file which was done in the upstream grib2_0p25 job, fix bugzilla 1226
+    # "${DBNROOT}/bin/dbn_alert" MODEL WAFS_0P25_UBL_GB2 "${job}" "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2"
+    "${DBNROOT}/bin/dbn_alert" MODEL WAFS_0P25_UBL_GB2_WIDX "${job}" "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2.idx"
+else
+    # retrieve UK products
+    # Three(3) unblended UK files for each cycle+fhour: icing, turb, cb
+    cat "${COMINuk}/egrr_wafshzds_unblended_"*"_0p25_${PDY:0:4}-${PDY:4:2}-${PDY:6:2}T${cyc}:00Z_t${fhr}.grib2" >"EGRR_WAFS_0p25_unblended_${PDY}_${cyc}z_t${fhr}.grib2"
+
+    if [[ "${MISSING_US_WAFS}" == "YES" ]]; then
+	if [ $SENDDBN = "YES" ] ; then
+	    "$DBNROOT/bin/dbn_alert" MODEL WAFS_UKMET_0P25_UBL_GB2 "${job}" "EGRR_WAFS_0p25_unblended_${PDY}_${cyc}z_t${fhr}.grib2"
+	fi
+    else
+	# pick up US data
+	cpreq "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2" .
+
+	# run blending code
+	export pgm="wafs_blending_0p25.x"
+
+	. prep_step
+
+	${EXECwafs}/${pgm} "WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2" \
+		   "EGRR_WAFS_0p25_unblended_${PDY}_${cyc}z_t${fhr}.grib2" \
+		   "0p25_blended_${PDY}${cyc}f${fhr}.grib2 >f${fhr}.out"
+
+	err=$?
+	if (( err != 0 )); then
+	    echo "turning back on dbn alert for unblended US WAFS product"
+	    "${DBNROOT}/bin/dbn_alert" MODEL WAFS_0P25_UBL_GB2_WIDX "${job}" "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2.idx"
+	    echo "WAFS blending 0p25 program failed at " ${PDY}${cyc}F${ffhr} > ../no_blending_files.$fhr
```

Comment:
There is no variable named `ffhr`
```suggestion
	    echo "WAFS blending 0p25 program failed at ${PDY}${cyc}F${fhr}" > "../no_blending_files.${fhr}"
```

Reply:
Thank you!